pkg/cluster/update.go

```diff
@@ -138,11 +156,15 @@ func getCount(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) in
 
 func (u *simpleUpgrader) listVMs(ctx context.Context, cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) ([]compute.VirtualMachineScaleSetVM, error) {
 	vmPages, err := u.vmc.List(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), "", "", "")
+	var vms []compute.VirtualMachineScaleSetVM
 	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
```

Comment:
This is a typed error; cast it to the type returned by List and read the error message correctly.

Reply:
I'm worried why this change is here - does the suggestion below of calling Deploy, not Upgrade, to do the restore mean this can be taken out?

Reply:
it's because we need to delete the whole scaleset prior to running the update, so the "list" op can't find the scaleset